src/net/java/sip/communicator/impl/protocol/jabber/ProtocolProviderServiceJabberImpl.java

```diff
@@ -1195,6 +1181,16 @@ private ConnectState connectAndLogin(
                 }
 
                 confConn.setCustomSSLContext(sslContext);
+                confConn.setHostnameVerifier(new HostnameVerifier()
+                {
```

Comment:
Why do we need this?

Reply:
Because Smack sets a default verifier which trows in case of a name mismatch:
https://github.com/igniterealtime/Smack/blob/master/smack-tcp/src/main/java/org/jivesoftware/smack/tcp/XMPPTCPConnection.java#L812

We don't want that, our verification service already validates the name (and allows to override a mismatch with the gui).